services/wallet/history/service.go

```diff
@@ -0,0 +1,339 @@
+package history
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/ethereum/go-ethereum/log"
+
+	statustypes "github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/multiaccounts/accounts"
+	statusrpc "github.com/status-im/status-go/rpc"
+	"github.com/status-im/status-go/rpc/network"
+
+	"github.com/status-im/status-go/services/wallet/chain"
+	"github.com/status-im/status-go/services/wallet/token"
+	"github.com/status-im/status-go/services/wallet/walletevent"
+)
+
+// EventBalanceHistoryUpdateStarted and EventBalanceHistoryUpdateDone are used to notify the UI that balance history is being updated
+const (
+	EventBalanceHistoryUpdateStarted           walletevent.EventType = "wallet-balance-history-update-started"
+	EventBalanceHistoryUpdateFinished          walletevent.EventType = "wallet-balance-history-update-finished"
+	EventBalanceHistoryUpdateFinishedWithError walletevent.EventType = "wallet-balance-history-update-finished-with-error"
+
+	balanceHistoryUpdateInterval = 12 * time.Hour
+)
+
+type Service struct {
+	balance        *Balance
+	db             *sql.DB
+	eventFeed      *event.Feed
+	rpcClient      *statusrpc.Client
+	networkManager *network.Manager
+	tokenManager   *token.Manager
+	serviceContext context.Context
+	cancelFn       context.CancelFunc
+
+	timer          *time.Timer
+	visibleSymbols []string
+	mutex          sync.Mutex // Protects access to visibleSymbols
```

Comment:
nit: it seems the "good" practice is to reuse variable name + mutext e.g `visibleSymbolsMutex`

For consistency maybe name visibleSymbols: visibleTokenSymbols